test/test.getch.ts

```diff
@@ -1102,3 +1102,236 @@ describe('🍂 defaults & instances', () => {
+      await instance.request({url});
+      scope.done();
+      assert.strictEqual(spyFunc.callCount, 1);
+    });
+
+    it('should invoke multiple request interceptors in the order they were added', async () => {
+      const scope = nock(url)
+        .matchHeader('foo', 'bar')
+        .matchHeader('bar', 'baz')
+        .matchHeader('baz', 'buzz')
+        .get('/')
+        .reply(200, {});
+      const instance = new Gaxios();
+      instance.interceptors.request.addInterceptor({
+        resolved: config => {
+          config.headers!['foo'] = 'bar';
+          return Promise.resolve(config);
+        },
+      });
+      instance.interceptors.request.addInterceptor({
+        resolved: config => {
+          assert.strictEqual(config.headers!['foo'], 'bar');
+          config.headers!['bar'] = 'baz';
+          return Promise.resolve(config);
+        },
+      });
+      instance.interceptors.request.addInterceptor({
+        resolved: config => {
+          assert.strictEqual(config.headers!['foo'], 'bar');
+          assert.strictEqual(config.headers!['bar'], 'baz');
+          config.headers!['baz'] = 'buzz';
+          return Promise.resolve(config);
+        },
+      });
+      await instance.request({url, headers: {}});
+      scope.done();
+    });
+
+    it('should not invoke a any request interceptors after they are removed', async () => {
+      const scope = nock(url).persist().get('/').reply(200, {});
+      const spyFunc = sinon.fake(
+        () =>
+          Promise.resolve({
+            url,
+            validateStatus: () => {
+              return true;
+            },
+          }) as unknown as Promise<GaxiosOptions>
+      );
+      const instance = new Gaxios();
+      instance.interceptors.request.addInterceptor({
+        resolved: spyFunc,
+      });
+      instance.interceptors.request.addInterceptor({
+        resolved: spyFunc,
+      });
+      instance.interceptors.request.addInterceptor({
+        resolved: spyFunc,
+      });
+      await instance.request({url});
+      instance.interceptors.request.removeAll();
+      await instance.request({url});
+      scope.done();
+      assert.strictEqual(spyFunc.callCount, 3);
+    });
+
+    it('should invoke the rejected function when a previous request interceptor rejects', async () => {
+      const instance = new Gaxios();
+      instance.interceptors.request.addInterceptor({
+        resolved: () => {
+          throw new Error('Something went wrong');
+        },
+      });
+      instance.interceptors.request.addInterceptor({
+        resolved: config => {
+          config.headers = {hello: 'world'};
+          return Promise.resolve(config);
+        },
+        rejected: err => {
+          assert.strictEqual(err.message, 'Something went wrong');
+        },
+      });
+      // Because the options wind up being invalid the call will reject with a URL problem.
+      assert.rejects(instance.request({url}));
```

Comment:
Do we need an `await` here if we make the calls between `request` and the interceptor calls async?

```suggestion
      assert.rejects(await instance.request({url}));
```

Reply:
I don't believe so `assert.rejects` will await the promise returning function.